duckduckgo/BrowserServicesKit
# Add PrivacyConfig subfeatures + remote internal user feature flagging

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/72649045549333/1203676877763836
iOS PR:  
macOS PR: 
What kind of version bump will this require?: Major/Minor/Patch

**Optional**:

Tech Design URL:
https://app.asana.com/0/481882893211075/1203938569856702
https://app.asana.com/0/481882893211075/1203984214904257
CC: 

**Description**:

This PR adds subfeatures support to the Privacy Config component, adds support for an additional state, internal, and moves the InternalUserDecider from iOS into BSK to be used by the `AppPrivacyConfiguration` to provide the correct state for internal users.

I'm mainly opening this as a draft to get some initial eyes on it before opening a final PR. It needs more testing (automated and manual). I'm going to be separating the subfeature part out in parallel as this is more pressing for the Autofill project.



**Steps to test this PR**:
1. Check out the iOS PR above
2. Check that any existing features configured using the privacy-config still work as expected



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
